onnx/defs/rnn/defs.cc

```diff
@@ -223,6 +225,11 @@ Equations (Default: f=Sigmoid, g=Tanh):
           "for default if not specified.",
           AttributeProto::STRINGS,
           OPTIONAL)
+    .Attr("linear_before_reset", "When computing the output of the hidden gate, "
+          "apply the linear transformation before multiplying by the output of the "
+          "reset gate.",
+          AttributeProto::INT,
+          OPTIONAL)
```

Comment:
Please don't use this "OPTIONAL". Using the API to feed default value for optional attribute. Thanks!

Reply:
Dumb question, any existing model applying this mechanism (apply linear transformation before multiplying) with better results please? Just want to understand more about the benefit of introducing this new attribute please.

Reply:
Version needs to be bumped in this case. You may refer to PR #390 for bumping the version.

Reply:
You'd need to add SinceVersion() here and add the 'old.cc' file similarly to #390

Reply:
@linkerzhang - I'm not sure whether it leads to better results, but the issue is that this is the version which CuDNN (and subsequently PyTorch) implements. I guess it's better to keep both versions representable (though not necessarily supported by all backends)

Reply:
@dzhulgakov Thank you very much for the clarification! Agree the point.